utils/rowgroup/rowgroup.cpp

```diff
@@ -993,8 +979,8 @@ bool Row::equals(const Row& r2, uint32_t lastCol) const
   // hasCollation is true if there is any column of type CHAR, VARCHAR or TEXT
   // useStringTable is true if any field declared > max inline field size, including BLOB
   // For memcmp to be correct, both must be false.
-  if (!hasCollation && !useStringTable && !r2.hasCollation && !r2.useStringTable)
-    return !(memcmp(&data[offsets[0]], &r2.data[offsets[0]], offsets[lastCol + 1] - offsets[0]));
+  //if (!hasCollation && !useStringTable && !r2.hasCollation && !r2.useStringTable)
```

Comment:
I strongly disagree with this comment. This should work.

Reply:
It will not for inline stored strings, with byte sizes 9..20. The NULL marks for them are stored at the end and value is not changed when string field is set to a NULL value.

That means that if you set string field to "a" and then to NULL, the data at appropriate colIndex will still be "a".

This, in turn, means that if you construct rows A and B by setting inline string field to "a" and then to NULL for A and setting string field to "b" and then to NULL for B, the memcmp will consider them different despite the fact they are the same.

So i strongly disagree with your disagreement. The memcmp will not work.

Reply:
I think I've found the way for this to work. It is in testing now.

Reply:
It works.